### PR TITLE
Add docs on how to keep this gem in sync with the Design System

### PR DESCRIPTION
**Pull request by alex-ju**
## What
Add docs on how to keep this gem in sync with the Design System.

## Why
As agreed at the community meeting on 21 October.

## Visual Changes
No visual changes.

Please be picky on the wording and come up with suggestions.

### CODE REVIEW

File: docs/upgrade-govuk-frontend.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+3. Check if and how the new feature changes affect this repository. If a new feature is added to a component, check the markup of the component associated with the feature mirrors the one in `govuk-frontend`. We usually benefit from the JavaScript and SCSS updates without requiring any changes, however make sure these are being imported in the component as expected. If a feature was pushed upstream from this gem to the Design System make sure to update the code to use the feature from upstream and remove the unused code in this gem. We decided that we do not bring features in `govuk_publishing_components` unless there is a user need for this (for example it improves the accessibility of a component) or it makes the code more robust (for example it provides better browser support).
```
Comment (maxgds) [inline docs/upgrade-govuk-frontend.md:12]: Nitpick, English usage: "we do not bring features in" in should be "into". I think it also works better as "We decided that we **will not** bring features into"

**Review by maxgds — CHANGES_REQUESTED** [review]
Looks good as is, but I have added one small suggestion.

Additionally a bit of guidance on what the version for this gem should be at the end of this might be good. If a breaking change comes in from govuk-frontend it might be tempting to mark this as breaking, but that's probably not usually necessary.

**Comment by alex-ju** [comment]
> Additionally a bit of guidance on what the version for this gem should be at the end of this might be good. If a breaking change comes in from govuk-frontend it might be tempting to mark this as breaking, but that's probably not usually necessary.

Indeed the `govuk-frontend` release type has no direct relationship with this gem's release type. I added a phrase to state that – open to suggestions on the wording as always.

**Review by maxgds — APPROVED** [review]
I am happy with this. Would like to see a second opinion, ideally though.

File: docs/upgrade-govuk-frontend.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+1. Read the [release notes](https://github.com/alphagov/govuk-frontend/releases/). GOV.UK Frontend has well-documented release notes that, depending on the release type, might contain the following sections:
+ - Breaking changes (major release)
+ - New features (minor release)
+ - Fixes (patch release)
+
+2. Check if and how the breaking changes affect this gem and GOV.UK applications dependent on this gem. For each breaking change follow the migration steps in the release notes and push the changes in a separate commit on the same Dependabot branch. Make sure to test GOV.UK applications with this branch. On major releases is always a good idea to involve someone from [the Design System team](https://github.com/orgs/alphagov/teams/team-gov-uk-design-system/members) in the review process as they have a thorough understanding of the changes.
```
Comment (owenatgov) [inline docs/upgrade-govuk-frontend.md:10]: I see what you've done here with using points 2, 3 and 4 to reflect breaking changes, new features and fixes respectively, however it took me a moment to realise that's what the intent was and I got a bit confused going from 2 to 3. I wonder if these paragraphs would benefit being under a single list item and separated by headings ("For any breaking changes in this release..."). I appreciate that that's going to break markdown formatting and might generally be a faff to make look nice so I'm not precious about this.

**Review by owenatgov — APPROVED** [review]
Clear and concise IMO 👌 I've noted one preferential formatting point but it's not a blocker for getting this merged.

File: docs/upgrade-govuk-frontend.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Keep this gem in sync with the Design System
+
```
Comment (chris-gds) [inline docs/upgrade-govuk-frontend.md:2]: So what confused me when I started was how the Design System (& the team) links into this gem and in what way.  I wasn't sure what came first (the Design System or the gem), what should take precedence (eg the DS is _always_ the single point of truth) or where the line is between how much development can take place within this app before the DS should be involved.

Not sure if this is apt for this PR but i thought a one-liner might maker it clearer.

Eg something like:
The [Design System](https://design-system.service.gov.uk/) team are part of GDS (not GOVUK) their work heavily informs the code within this gem to create a set of UI components. This  a gem aims to document and distribute components in GOV.UK [frontend applications](https://docs.publishing.service.gov.uk/#frontend-apps)

File: docs/upgrade-govuk-frontend.md (hunk 2 above)
Comment (alex-ju) [reply docs/upgrade-govuk-frontend.md]: I see your point, Owen. I have restructured the documentation into 3 main steps in the last commit. Let me know if this addresses your point.

File: docs/upgrade-govuk-frontend.md (hunk 3 above)
Comment (alex-ju) [reply docs/upgrade-govuk-frontend.md:2]: Fair point, Chris. The paragraph in the main README file does not contain much info around this, and people may miss it. I rephrased the first part of this doc trying to provide this context in the last commit.

File: docs/upgrade-govuk-frontend.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+5. Add a final commit with a [CHANGELOG](/CHANGELOG.md) entry mirroring the pull request name (for example "Bump govuk-frontend from 3.8.1 to 3.9.1").
+
+6. If the pull request required any code changes ask a colleague for a review, otherwise you can approve it and merge it.
+
+Note that the `govuk-frontend` release type does not dictate the `govuk_publishing_components` release type the latter being based on the required changes imposed by the former.
+
+See an [example pull request](https://github.com/alphagov/govuk_publishing_components/pull/1705) for upgrading to a new minor version of `govuk-frontend`.
+
+See an [example pull request](https://github.com/alphagov/govuk_publishing_components/pull/1010) for upgrading to a new major version of `govuk-frontend`.
```
[the remark this answers is not in the document]
Comment (alex-ju) [reply docs/upgrade-govuk-frontend.md:42]: Hmm, I'm not sure I get this question. The major version _is_ a breaking change. We know fixes (patch releases) of `govuk-frontend` do not require any changes, hence not much to share for this case – it's a simple version bump.

**Comment by alex-ju** [comment]
Thanks to everyone involved for reviewing this. Have tried to address the comments and rebased to master.

File: docs/upgrade-govuk-frontend.md (hunk 2 above)
Comment (owenatgov) [reply docs/upgrade-govuk-frontend.md]: Exceptional. Cheers Alex.

**Comment by alex-ju** [comment]
I sense that people interested in this topic are fine with this, so I'm going to merge it. Thanks to all involved in shaping this up!

**Pull request merged**